chingu-voyages/v52-tier3-team-38
# Gary/feature/appointment pagination

# Description
This PR finalizes pagination for the list view of appointment requests. It also implements a mark as visited button that calls a PUT request to update the status of an appointment from pending to visited. Lastly, The admin navbar can be used to navigate toward their respective pages.

## Related to the following Jira/Trello issue 
https://trello.com/c/SlwMJ1LH/39-view-resident-requests-list-view

## Main changes explained:
- Refactored pagination of data to have it be handled in a server component instead of using react-bootstrap.
- Created MarkVisitedButton component to call api to update appointment status information. 
- Admin Navbar was adjusted to link to their own pages.

## How to test:
1. check into current branch
2. do `npm install` and `npm run dev` to run this PR locally
3. go to `include url to view your component/feature` 
4. Login as an admin and navbar to go to the requests page. You should see all current requests, some pending, some visited.
5. Test that the pagination is correctly implemented, it should be able to go to the next page to the next set of data, as well as go back to the previous page to see prior data.
6. Test marking a appointment as visited. The appointment will be moved to the last page after its status was updated, verify that the updated appointments status is visited.

## Note:
Due to time constraints, I only did what the ticket acceptance criteria required.
